app/src/main/java/org/simple/clinic/teleconsultlog/shareprescription/TeleconsultSharePrescriptionModel.kt

```diff
@@ -46,4 +48,8 @@ data class TeleconsultSharePrescriptionModel(
   fun medicalRegistrationIdLoaded(medicalRegistrationId: String): TeleconsultSharePrescriptionModel {
     return copy(medicalRegistrationId = medicalRegistrationId)
   }
+
+  fun downloadButtonStateChanged(downloadButtonState: ButtonState?): TeleconsultSharePrescriptionModel {
```

Comment:
Can we have two separate methods like `downloading` and `downloadCompleted` to update the button state in model.

Reply:
We are handling the button states using a single function across the project.

Reply:
We are not actually using it across entire project. It's preferable to have individual methods to update the button state instead of passing them down from the update function. Also having individual functions like `downloading` & `downloadCompleted` will be much more readable imo. WDYT?